internal/zero/api/api.go

```diff
@@ -34,6 +37,13 @@ const (
 	minTelemetryTokenTTL = time.Minute * 5
 )
 
+// see https://github.com/pomerium/pomerium-zero/issues/1711
+var connectClientKeepaliveParams = keepalive.ClientParameters{
+	Time:                time.Minute, // send pings every minute
+	Timeout:             time.Minute, // wait 1 minute for ping ack
+	PermitWithoutStream: false,
```

Comment:
Shouldn't this be `true`?

> // If true, client sends keepalive pings even with no active RPCs. If false,
> // when there are no active RPCs, Time and Timeout will be ignored and no
> // keepalive pings will be sent.

Reply:
couple reasons: 

1. we are talking to GLB rather then to our server app, and it may not like it (I've seen it shut down the connection if keep-alive is too aggressive, so it definitely has some _unknown_ keep-alive enforcement policy in place); thus the choice of very conservative 1 min interval. 
2. there is a control loop to restart `Subscribe()` if it fails,